server/src/main/java/io/helidon/extensions/mcp/server/McpServerFeature.java

```diff
@@ -992,4 +1005,63 @@ private McpFeatures mcpFeatures(JsonRpcRequest req,
         }
         return session.createFeatures(requestId);
     }
+
+    private Optional<JsonRpcError> throwableExceptionHandler(JsonRpcRequest request,
+                                                             JsonRpcResponse response,
+                                                             Throwable throwable) {
+        return sendError(request, response, throwable, INTERNAL_ERROR);
+    }
+
+    private Optional<JsonRpcError> mcpExceptionHandler(JsonRpcRequest request,
+                                                       JsonRpcResponse response,
+                                                       Throwable throwable) {
+        if (throwable instanceof McpException exception) {
+            return sendError(request, response, exception, exception.code());
+        }
+        return throwableExceptionHandler(request, response, throwable);
+    }
+
+    private Optional<JsonRpcError> sendError(JsonRpcRequest request,
+                                             JsonRpcResponse response,
+                                             Throwable throwable,
+                                             int errorCode) {
+        var session = findSession(request);
+        if (session.isEmpty()) {
+            response.status(Status.NOT_FOUND_404).send();
+            return Optional.empty();
+        }
+        JsonValue requestId = request.rpcId().orElse(JsonValue.NULL);
+        Optional<McpFeatures> features = session.get().features(requestId);
+        if (features.isEmpty()) {
+            response.status(Status.NOT_FOUND_404).send();
+            return Optional.empty();
+        }
+        response.error(errorCode, throwable.getMessage());
+        var sseSink = features.get().sseSink().orElse(null);
```

Comment:
Unfortunately, for subscribe/unsubscribe there's another `SseSink` in play. Error handling may need to be special there. Maybe just file an issue and I can look at this case. Not as urgent IMO.

Reply:
Right, I did not address that change in the last commit.

Reply:
See https://github.com/helidon-io/helidon-mcp/issues/71